### PR TITLE
cargo-information: init at 0.4.2

### PR DESCRIPTION
**Pull request by eopb**
## Description of changes

Added new package [`cargo-information`](https://github.com/hi-rustin/cargo-information).
`cargo-information` is a tool for rust developers conveniently prints details about rust packages.

## Things done

Tested on NixOS:

```
nixpkgs on  init-cargo-information [+] took 11s
❯ : nix run .#cargo-information -- info redact
    Updating crates.io index
redact #secret #logging #pii #password #no_std
A simple library for keeping secrets out of logs.
version: 0.1.9
license: MIT/Apache-2.0
rust-version: 1.65
documentation: https://docs.rs/redact/
repository: https://github.com/eopb/redact
features:
  default = [std]
  std     = []
  fake    = [dep:fake, dep:rand]
  serde   = [dep:serde]
  zeroize = [dep:zeroize]
dependencies:
  fake@2.5
  rand@0.8
  serde@1.0
  zeroize@1.7
owners:
  eopb (Ethan Brierley)
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ca/cargo-information/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "cargo-test-macro-0.1.0" = "sha256-4u3Ium+WYBdyocuehDulRgUOR74JC6AUI2+A5xlnUGw=";
+    };
+  };
+
+  # The tests require network access which is not available in sandboxed Nix builds.
+  doCheck = false;
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-information/package.nix:29]: Looks like not all tests require network. Maybe a
```nix
  checkFlags = [
    # reason for disabling test
    "--skip=example::tests:example_test"
  ];
```
is better.

File: pkgs/by-name/ca/cargo-information/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  meta = with lib; {
+    description =  "A cargo subcommand to show information about crates";
+    mainProgram = "cargo-info";
+    homepage = "https://github.com/hi-rustin/cargo-information";
+    changelog = "https://github.com/hi-rustin/cargo-information/blob/v${src.rev}/CHANGELOG.md";
+    license = with licenses; [ mit ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-information/package.nix:46]:
```suggestion
    license = licenses.mit;
```

**Comment by eopb** [comment]
Thank you for your review comments @Aleksanaa 

I've force pushed, addressing your comments

File: pkgs/by-name/ca/cargo-information/package.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  ];
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  meta = with lib; {
+    description =  "A cargo subcommand to show information about crates";
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-information/package.nix:47]:
```suggestion
    description = "A cargo subcommand to show information about crates";
```

File: pkgs/by-name/ca/cargo-information/package.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "cargo-test-macro-0.1.0" = "sha256-4u3Ium+WYBdyocuehDulRgUOR74JC6AUI2+A5xlnUGw=";
+    };
+  };
+
+  checkFlags = [
+    # Some tests require network access which is not available in sandboxed Nix builds.
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-information/package.nix:29]:
```suggestion
    # Require network access
```

Comment (eopb) [reply pkgs/by-name/ca/cargo-information/package.nix]: :+1: that's better, https://github.com/NixOS/nixpkgs/compare/b2f3a820395a928da4bb35326a47eb5ecde55483..62be6e81405cfd2dd47caf7074f297995b20618c

File: pkgs/by-name/ca/cargo-information/package.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-information/package.nix:45]:
```suggestion
postFixup = ''
  wrapProgram $out/bin/cargo-info \
    --prefix PATH : ${lib.makeBinPath [ rustc ]}
'';
```

```
=> cargo-info info libadwaita
error: could not execute process `rustc -vV` (never executed)

Caused by:
  No such file or directory (os error 2)
```

**Comment by Aleksanaa** [comment]
Tested

```
=> cargo-info info libadwaita
    Updating crates.io index
  Downloaded libadwaita v0.6.0
  Downloaded 1 crate (108.1 KB) in 2.62s
libadwaita #gtk4-rs #gnome #adwaita #GUI
Rust bindings for libadwaita
version: 0.6.0
license: MIT
rust-version: unknown
documentation: https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita
homepage: https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/
repository: https://gitlab.gnome.org/World/Rust/libadwaita-rs
features:
  gtk_v4_10 = [gtk_v4_8]
  gtk_v4_8  = [gtk_v4_6]
  gtk_v4_6  = [gtk_v4_4]
  gtk_v4_4  = [gtk_v4_2]
  gtk_v4_2  = []
  gtk_v4_12 = [gtk_v4_10]
  v1_1      = [ffi/v1_1]
  v1_2      = [v1_1, ffi/v1_2]
  v1_3      = [v1_2, ffi/v1_3]
  v1_4      = [v1_3, ffi/v1_4]
  v1_5      = [v1_4, ffi/v1_5]
dependencies:
  libadwaita-sys@0.6
  gdk4@0.8
  gdk-pixbuf@0.19
  gio@0.19
  glib@0.19
  gtk4@0.8
  libc@0.2
  pango@0.19
```

File: pkgs/by-name/ca/cargo-information/package.nix (hunk 5 above)
Comment (eopb) [reply pkgs/by-name/ca/cargo-information/package.nix:46]: Thank you https://github.com/NixOS/nixpkgs/compare/62be6e81405cfd2dd47caf7074f297995b20618c..0feae8bf3ccc1945d85f6a7d688242161ed6fd7a

I managed to reproduce this by creating a pure shell with no packages and running the binary there. Do you know if there's also an easier way to do it, perhaps a flag I can pass to `nix run .#cargo-information`?

Comment (eopb) [reply pkgs/by-name/ca/cargo-information/package.nix:46]: Oops, I missed adding `makeWrapper` to my inputs. I'll push a fix

Comment (eopb) [reply pkgs/by-name/ca/cargo-information/package.nix:48]: https://github.com/NixOS/nixpkgs/compare/0feae8bf3ccc1945d85f6a7d688242161ed6fd7a..fba585953a0b8211c5ae81fd6ed777ab81d2a335

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**